server/src/lib/handlers/intermediate-file-validator.ts

```diff
@@ -59,66 +61,48 @@ export function fileValidator(req: Request, res: Response, uniqueSlugs: Set<stri
   req.pipe(busboy)
 }
 
-async function validateByKey(s3:any, data: any, type: string, uniqueSlugs: Set<string>) {
-  const { Name, Contents } = data
-  let result: {[key: string]: any} | any = { exceptions: [] }
-  for(const file of Contents) {
-    const key = file.Key
-    try {
-      const readableStream = s3.getObject({
-      Bucket: Name,
-      Key: key 
-    })
-    .createReadStream()
-    result = await asyncValidateStream(readableStream, type, result, uniqueSlugs)
-    } catch(error) {
-      const errorKey = result.exceptions.find((err: { key: string; }) => err.key === error)
-
-      if(!errorKey){
-        result.exceptions.push({
-          key: error,
-          ids: [key]
-        })
-      } else {
-        errorKey.ids.push(key)
-      }
-    }
-  }
-  return result
+export async function s3keyValidator(req: Request, res: Response, uniqueSlugs: Set<string>,workerPool:WorkerThreadPool): Promise<Response | void> {
+  const { type, data } = req.body
+  const result = await validateS3Files(data, type, uniqueSlugs,workerPool);
+  res.json(result)
+  return
 }
-export async function s3keyValidator(req: Request, res: Response, uniqueSlugs: Set<string>): Promise<Response | void> {
+
+export function getFiles(req: Request, res: Response){
   const { type, data: path } = req.body
   const s3keyParts = path.split('/')
-  const bucket = s3keyParts[2]
-  const keyPrefix = s3keyParts.slice(3).join('/') + '/' + type.toLowerCase()
   const s3 = new S3({
     accessKeyId: config['accessKeyId'],
     secretAccessKey: config['secretAccessKey']
   })
-
+  const bucket = s3keyParts[2]
+  const keyPrefix = s3keyParts.slice(3).join('/') + '/' + type.toLowerCase()
```

Comment:
This does not look like a safe way to parse the S3 details, please check if any AWS npm packages support this feature, Ideally there should be something that should be already available

Reply:
@Robert-Christopher as discussed, we will be going ahead with the existing code as it is mostly being used internally